packages/lodestar/src/chain/chain.ts

```diff
@@ -338,7 +327,7 @@ export class BeaconChain extends (EventEmitter as { new(): ChainEventEmitter })
     await this.db.setChainHeadRoots(blockRoot, block.stateRoot);
     this.forkChoice.addBlock(block.slot, blockRoot, block.parentRoot, newState.currentJustifiedCheckpoint,
       newState.finalizedCheckpoint);
-    // await this.applyForkChoiceRule();
+    await this.applyForkChoiceRule();
     await this.updateDepositMerkleTree(newState);
```

Comment:
Do we need to handle different versions of the deposit tree in different forks?

Reply:
What I see is fork choice spec does not mention about deposit tree.

Right now we get state of the parent block and process this block into `newState`. So I think storing deposit tree in this `newState` already has the meaning of "fork" here.

Not sure if that's your concern.

Reply:
Oh I found out that we use same merkleTree database in `updateDepositMerkleTree`  which may be an issue, will look into that..

Reply:
Probably we don't have any issues here.
All deposits are sorted, given 2 different states respective to 2 different forked blocks, as long as they have the same `eth1DepositIndex`, `merkleTree` would be the same.